hydra-node/src/Hydra/HeadLogic.hs

```diff
@@ -293,22 +387,10 @@ update Environment{party, signingKey, otherParties} ledger st ev = case (st, ev)
         Valid -> [ClientEffect $ TxValid tx, NetworkEffect $ ReqTx party tx]
         Invalid err -> [ClientEffect $ TxInvalid{utxo = utxo, transaction = tx, validationError = err}]
   (OpenState{parameters, coordinatedHeadState = headState@CoordinatedHeadState{seenTxs, seenUTxO}, previousRecoverableState}, NetworkEvent (ReqTx _ tx)) ->
-    case applyTransactions ledger seenUTxO [tx] of
-      Left (_, err) -> Wait $ WaitOnNotApplicableTx err -- The transaction may not be applicable yet.
-      Right utxo' ->
-        let newSeenTxs = seenTxs <> [tx]
-         in nextState
-              ( OpenState
-                  { parameters
-                  , coordinatedHeadState =
-                      headState
-                        { seenTxs = newSeenTxs
-                        , seenUTxO = utxo'
-                        }
-                  , previousRecoverableState
-                  }
-              )
-              [ClientEffect $ TxSeen tx]
+    processL2OutCome l2Outcome (Just parameters) previousRecoverableState
+   where
+    eitherTx = applyTransactions ledger seenUTxO [tx]
+    l2Outcome = onOpenHandleNetworkReqTxEvent tx headState seenTxs eitherTx
```

Comment:
In here it's very important that `applyTransactions ledger seenUTxO [tx]` is IN the part that is documented. That is the "meat" of the function!